uaclient/event_logger.py

```diff
@@ -42,8 +42,8 @@ class EventLoggerMode(enum.Enum):
 
 class EventLogger:
     def __init__(self):
-        self._error_events = []  # type: List[Dict[str, Optional[str]]]
-        self._warning_events = []  # type: List[Dict[str, Optional[str]]]
+        self._error_events = []  # type: List[Dict[str, Any]]
+        self._warning_events = []  # type: List[Dict[str, Any]]
```

Comment:
Using `Any` in types makes me sad :( but I understand what is happening.

Would it be too much to define a simple class just to handle the different fields here instead of using a dict?

Reply:
Related but not necessarily useful for this case. I'd like to add a `to_dict()` method to `uaclient.data_types.DataObject` to have simple dict->PythonObject->dict translation.

Reply:
Since we will be using a non-nested approach and also assuming all values will strings, I can turn this back into `List[Dict[str, str]]`. But if that changes in the future, the class idea is a good one

Reply:
I have not created a class for it, but a `Union` type, specially since we will not be using a nested dict anymore. Also, I think we can always use string as values for the `additional_info` fields, which will simplify the logic.

What do you think ?